vignettes/auth.Rmd

```diff
@@ -12,104 +12,76 @@ vignette: >
+
+-   Bot authentication allows you to create a fully automated Twitter bot that performs actions on its own behalf rather than on behalf of a human.
+
+In either case, you'll need to start by creating a Twitter app, so this vignette will first walk you through that process.
+You'll then learn how to create and save the objects needed to use these two new types of authentication.
 
 ```{r}
-rtweet::get_token()
+library(rtweet)
+```
+
+## Creating a Twitter app
+
+To create a Twitter app, you need to first apply for a developer account by following the instructions at <https://developer.twitter.com>.
+Once you have acquired a developer account, navigate to the [developer portal](https://developer.twitter.com/en/portal/projects-and-apps) and click the "Create App" button at the bottom of the page.
+You'll need to name your app: the name is unimportant for our purposes, but needs to be unique across all twitter apps.
+
+Once you've created your app, click on "Keys and tokens" to get a screen like this:
+
+![](keys-tokens.png){width="362"}
+
+You'll need some data from this screen in the next step.
+
+## Setup
+
+Now you're ready to tell rtweet how to authenticate using your app.
+You have two options:
+
+-   `rtweet_user()` is best for collecting data.
+-   `rtweet_bot()` is best for creating twitter bots.
+
+In both cases, you first create an auth object using data from the "tokens & keys" page then save it for use in future sessions.
+This process is a little trickier than you might expect because you never want to enter API keys into the R console or save in an .R script to avoid accidentally exposing secrets.
+
+Start by running either of the two lines in your console:
+
+```{r, eval = FALSE}
+auth <- rtweet_app()
+auth <- rtweet_bot()
+```
+
+You will then get a series of pop-ups asking you to enter data from the "keys & tokens" page:
+
+-   `rtweet_app()` is easy: you need to enter the bearer token, which you can get by clicking the "regenerate" button next to "bearer token".
+
+-   `rtweet_bot()` is a bit more work because you need four pieces of data: the API key and secret, and the access token and secret.
+    To get these two pairs of data, you'll need to click the two regenerate buttons.
+
+(You may also want to record this information in another secure location; there's no other way to get it back except to regenerate, which will break any existing uses.)
+
+Next, check that you're entered everything correctly and your auth works:
+
+```{r, eval = FALSE}
+df <- search_tweets("#rstats", token = auth)
+```
+
+Assuming that's good, you can now run `auth_save()` to save your authentication object so that you can use it again in the future:
+
+```{r, eval = FALSE}
+auth_save(auth, "your-app-name")
 ```
 
-That's it!
+## Use
+
+Now, in a future session, you can use your saved auth by calling `auth_as()`:
```

Comment:
The language here makes it sound a little bit like you can only save app auth, since you recover an auth via the app name. I'm also curious how this works when all the auths come from the same app. In theory I could have one Twitter project with one app and use that same application name for all three types of authentication, if my understanding is correct. How does selecting amongst those strategies work here since there is only a single app name for three auth strategies in that case (again, subject to me understanding the Twitter API correctly)?

Reply:
You can save any auth object. The name is arbitrary, so presumably you'd use some suffix to distinguish them. Maybe I need to make that more clear?

Reply:
I think that would help. I think the instructions about how to use `rtweet` are fairly clear, but using the same app name for all three types of authentication might lead people in poor mental models of Twitter side authentication.